src/utils/MSstatsConverter.cpp

```diff
@@ -0,0 +1,419 @@
+        }
+
+        if (headers_valid)
+        {
+          // Map the column name to the index
+          for (Size j = 0; j < n_cols; ++j)
+          {
+            columnname_to_columnindex[line[j]] = j;
+          }
+          break;
+        }
+        else
+        {
+          for (auto const & entry: diff)
+          {
+            LOG_ERROR << " " << entry;
+          }
+          LOG_ERROR << std::endl;
+          return ILLEGAL_PARAMETERS;
+        }
+      }
+
+      // Iterate all remaining lines in the experimental design file
+      for (++i; i < n_lines; ++i)
+      {
+        std::vector< String > line;
+        file_experimental_design.getRow(i, line);
+
+        // Skip empty lines
+        if (line.empty())
+        {
+          continue;
+        }
+        // Check whether the number of entries in this line is as expected
+        if (line.size() != headers_size)
+        {
+          LOG_ERROR << "ERROR: Wrong number of entries in line "
+              << (i + 1) << ". Have: "  << line.size() << ". Expected: " << headers_size << std::endl;
+          return ILLEGAL_PARAMETERS;
+        }
+        String const & filename = line[columnname_to_columnindex[TOPPMSstatsConverter::msstats_header_filename]];
+        edkey_to_rowindex[filename].insert(i);
+      }
+    }
+
+    // Print column name to index mapping
+    if (this->debug_level_ > 0)
+    {
+      this->writeDebug_("Experimental Design Columns:", 1);
+      for (auto const & columnname : columnname_to_columnindex)
+      {
+        this->writeDebug_(columnname.first + " : " + columnname.second, 1);
+      }
+    }
+    // Read the input files
+    ConsensusMap consensus_map;
+    ConsensusXMLFile().load(this->getStringOption_(TOPPMSstatsConverter::param_in_consensusxml), consensus_map);
+
+    // The output file of the MSstats converter (TODO Change to CSV file once store for CSV files has been implemented)
+    TextFile csv_out;
+
+    // Add the header line
+    csv_out.addLine("ProteinName,PeptideSequence,PrecursorCharge,FragmentIon,ProductCharge,IsotopeLabelType,Condition,BioReplicate,Run,Intensity");
+
+    // Regex definition for fragment ions
+    std::regex regex_msstats_FragmentIon("[abcxyz][0-9]+");
+
+    // Iterate protein identifications and collect spectra_data metavalue and ID
+    std::map< String, std::set< String> > protid_to_filenames;
+    for (auto const & protein_identification : consensus_map.getProteinIdentifications())
+    {
+      String const & identifier = protein_identification.getIdentifier();
+
+      if (protein_identification.metaValueExists(TOPPMSstatsConverter::meta_value_exp_design_key) == false)
+      {
+        LOG_ERROR << "ERROR: ProteinIdentification does not have meta value for original file. Cannot continue!" << std::endl;
+        return ILLEGAL_PARAMETERS;
+      }
+
+      for (auto const & meta_value : protein_identification.getMetaValue(TOPPMSstatsConverter::meta_value_exp_design_key).toStringList())
```

Comment:
this line is a bit long and it's, thus, significantly harder to deduce the type of auto